vm/src/eval.rs

```diff
@@ -39,6 +41,12 @@ pub struct NexusVM<M: Memory> {
     ///
     /// Does include executing UNIMP instruction.
     pub max_trace_len: Option<usize>,
+
+    /// Profiling cycles
+    /// The global clk for execution trace.
+    pub global_clk: u64,
```

Comment:
This should maybe be like `cycle_count` or something, since instructions per clock cycle and the cycles needed for a specific instruction are not necessarily related 1-to-1 (yes the zkVM is at present single core and single threaded, but that could change). 

So we probably want to retain the notion of a clock as something distinct, in case we want to implement it more generally and thoroughly at some point.

Reply:
update to cycles count in 3abf38c